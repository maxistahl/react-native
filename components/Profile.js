import React from 'react';
import {Text, Image, View} from 'react-native';
import styles from './styles';
import Img from './Img';

export default class Profile extends React.Component {
  static navigationOptions = {
    title: 'Details'
  };

  render() {
    const {navigation} = this.props;
    const item = (navigation.getParam && navigation.getParam('item')) || this.props.item;

    return (
      <View style={styles.item}>
        <Img src={'http://tarmac.io/assets/members/' + item.pic + '.png'} style={{...styles.images, ...styles.big}} />
        <Text style={{...styles.text, ...styles.title}}>{item.name}</Text>
        <Text>{item.role}</Text>
        <Text>{item.description}</Text>
        <View style={{flex: 1, justifyContent: 'flex-start'}}>
          {item.github && (
            <View style={styles.textAndIcon}>
              <Image style={styles.icon} source={require('../assets/icon-github.png')} />
              <Text>{item.github}</Text>
            </View>
          )}
          {item.twitter && (
            <View style={styles.textAndIcon}>
              <Image style={styles.icon} source={require('../assets/icon-twitter.png')} />
              <Text>{item.twitter}</Text>
            </View>
          )}
        </View>
      </View>
    );
  }
}
