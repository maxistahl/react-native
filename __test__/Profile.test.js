/**
 * @jest-environment jsdom
 */
import React from 'react';
import {shallow} from 'enzyme';
import Profile from '../components/Profile';

const mock = {
  name: 'John Doe',
  role: 'Developer',
  pic: 'johndoe',
  github: 'https://github.com/',
  width: 500,
  height: 500,
  description: 'JohnDoe is a developer'
};

describe('Profile detail', () => {
  describe('Render', () => {
    const tree = shallow(<Profile item={mock} navigation={{navigate: true}} />);
    it('if the image uri provided is not available, render a default image', () => {
      expect(tree).toMatchSnapshot();
    });
  });
});
