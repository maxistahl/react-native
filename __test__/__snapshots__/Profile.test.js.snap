// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile detail Render if the image uri provided is not available, render a default image 1`] = `ShallowWrapper {}`;
